static/js/actions/index.js

```diff
@@ -124,18 +124,21 @@ export const receiveDashboardSuccess = programs => ({
   type: RECEIVE_DASHBOARD_SUCCESS,
   payload: { programs }
 });
-const receiveDashboardFailure = () => ({ type: RECEIVE_DASHBOARD_FAILURE });
+export const receiveDashboardFailure = errorInfo => ({
+  type: RECEIVE_DASHBOARD_FAILURE,
+  payload: { errorInfo }
+});
 export const clearDashboard = () => ({ type: CLEAR_DASHBOARD });
 
 export function fetchDashboard() {
   return dispatch => {
     dispatch(requestDashboard());
     return api.getDashboard().
       then(dashboard => dispatch(receiveDashboardSuccess(dashboard))).
-      catch(e => {
-        dispatch(receiveDashboardFailure());
+      catch(errorJson => {
```

Comment:
nit: JSON is the string representation of the object, not the object itself. Can we rename this to just `error` instead?